packages/opentelemetry-api/src/api/context.ts

```diff
@@ -78,12 +78,14 @@ export class ContextAPI {
    *
    * @param context context to be active during function execution
    * @param fn function to execute in a context
+   * @param args optional arguments forwarded to fn
    */
   public with<T extends (...args: unknown[]) => ReturnType<T>>(
     context: Context,
-    fn: T
+    fn: T,
+    ...args: unknown[]
```

Comment:
will investigate if better typing then `unknown[]` is possible here and adding `this`